domains/client/relayer/src/worker.rs

```diff
@@ -183,14 +202,22 @@ where
                 "Checking messages to be submitted from domain: {domain_id:?} at block: ({number:?}, {hash:?})",
             );
 
-            if let Err(err) = message_processor(relayer_id.clone(), &domain_client, hash) {
-                tracing::error!(
+            // check if the block is pruned
+            // if so, the node is lagging behind and there is no way to generate a proof
+            // mark this block processed and continue to next one.
+            if is_block_pruned(domain_id, number)? {
```

Comment:
The naming is misleading to me, technically it's the receipt that is pruned instead of the block.